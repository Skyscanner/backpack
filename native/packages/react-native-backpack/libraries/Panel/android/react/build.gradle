apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
