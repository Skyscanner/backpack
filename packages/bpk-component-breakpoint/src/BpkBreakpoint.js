/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import PropTypes from 'prop-types';
import React, { type Node } from 'react';
import MediaQuery from 'react-responsive';
import TOKENS from '@skyscanner/bpk-foundations-web/tokens/breakpoints.common';

const BREAKPOINTS = {
  SMALL_MOBILE: TOKENS.breakpointQuerySmallMobile,
  MOBILE: TOKENS.breakpointQueryMobile,
  SMALL_TABLET: TOKENS.breakpointQuerySmallTablet,
  SMALL_TABLET_ONLY: TOKENS.breakpointQuerySmallTabletOnly,
  TABLET: TOKENS.breakpointQueryTablet,
  TABLET_ONLY: TOKENS.breakpointQueryTabletOnly,
  ABOVE_MOBILE: TOKENS.breakpointQueryAboveMobile,
  ABOVE_TABLET: TOKENS.breakpointQueryAboveTablet,
};

type Props = {
  children: Node | ((matches: boolean) => mixed),
  query: string | $Values<typeof BREAKPOINTS>,
  legacy: boolean,
};

const BpkBreakpoint = (props: Props) => {
  const { children, query } = props;
  return <MediaQuery query={query}>{children}</MediaQuery>;
};

const queryValidator = (props: Props, ...rest) => {
  if (!props.legacy) {
    return PropTypes.oneOf(
      Object.keys(BREAKPOINTS).map((key) => BREAKPOINTS[key]),
    ).isRequired(props, ...rest);
  }
  return PropTypes.string.isRequired(props, ...rest);
};

BpkBreakpoint.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  query: queryValidator, // eslint-disable-line react/require-default-props
  legacy: PropTypes.bool, // eslint-disable-line react/no-unused-prop-types
};

BpkBreakpoint.defaultProps = {
  legacy: false,
};

export { BREAKPOINTS };
export default BpkBreakpoint;
