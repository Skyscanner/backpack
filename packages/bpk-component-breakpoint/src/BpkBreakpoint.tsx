/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { ReactNode } from 'react';
import MediaQuery from 'react-responsive';
// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import { breakpoints } from '@skyscanner/bpk-foundations-web/tokens/base.es6';

const BREAKPOINTS = {
  SMALL_MOBILE: breakpoints.breakpointQuerySmallMobile,
  MOBILE: breakpoints.breakpointQueryMobile,
  SMALL_TABLET: breakpoints.breakpointQuerySmallTablet,
  SMALL_TABLET_ONLY: breakpoints.breakpointQuerySmallTabletOnly,
  TABLET: breakpoints.breakpointQueryTablet,
  TABLET_ONLY: breakpoints.breakpointQueryTabletOnly,
  ABOVE_MOBILE: breakpoints.breakpointQueryAboveMobile,
  ABOVE_TABLET: breakpoints.breakpointQueryAboveTablet,
  ABOVE_DESKTOP: breakpoints.breakpointQueryAboveDesktop,
  DESKTOP_ONLY: breakpoints.breakpointQueryDesktopOnly,
} as const;

type Props = {
  children: ReactNode | ((matches: boolean) => ReactNode | null);
  query: string | (typeof BREAKPOINTS)[keyof typeof BREAKPOINTS];
  legacy?: boolean;
};

const BpkBreakpoint = ({ children, legacy = false, query }: Props) => {
  if (!legacy && !Object.values(BREAKPOINTS).includes(query)) {
    console.warn(
      `Invalid query ${query}. Use one of the supported queries or pass the legacy prop.`,
    );
  }
  return <MediaQuery query={query}>{children}</MediaQuery>;
};

export { BREAKPOINTS };
export default BpkBreakpoint;
