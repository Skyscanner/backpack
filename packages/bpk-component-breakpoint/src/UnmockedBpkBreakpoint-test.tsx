/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from '@testing-library/react';

import BpkBreakpoint, { BREAKPOINTS } from "./BpkBreakpoint";


describe('BpkBreakpoint', () => {
  it('should not render when breakpoint is not mocked within tests', () => {
    const { asFragment } = render(
      <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
        <div>doesnt match</div>
      </BpkBreakpoint>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
})