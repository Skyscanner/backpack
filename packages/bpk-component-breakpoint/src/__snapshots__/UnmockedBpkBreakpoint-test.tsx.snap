// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkBreakpoint should not render when breakpoint is not mocked within tests 1`] = `<DocumentFragment />`;
